openssl/src/sign.rs

```diff
@@ -100,7 +100,7 @@ impl<'a> Signer<'a> {
     /// OpenSSL documentation at [`EVP_DigestSignInit`].
     ///
     /// [`EVP_DigestSignInit`]: https://www.openssl.org/docs/manmaster/man3/EVP_DigestSignInit.html
-    pub fn new<T>(type_: MessageDigest, pkey: &'a PKeyRef<T>) -> Result<Signer<'a>, ErrorStack>
+    pub fn new<T>(type_: Option<MessageDigest>, pkey: &'a PKeyRef<T>) -> Result<Signer<'a>, ErrorStack>
```

Comment:
This is a breaking change. I'm not planning on a breaking release for about a year, FYI.

Reply:
Hm, would you prefer another solution then?
Like adding `new_without_digest`?

Reply:
Yeah something like that seems reasonable.